src/Security/Authenticator/GitHubAuthenticator.php

```diff
@@ -27,18 +27,14 @@
 
 class GitHubAuthenticator extends SocialAuthenticator
 {
-    private $clientRegistry;
-    private $em;
-    private $urlGenerator;
-    private $allowedGitHubOrganization;
     private $httpClient;
 
-    public function __construct(ClientRegistry $clientRegistry, ManagerRegistry $registry, UrlGeneratorInterface $urlGenerator, string $allowedGitHubOrganization)
-    {
-        $this->clientRegistry = $clientRegistry;
-        $this->em = $registry->getManager();
-        $this->urlGenerator = $urlGenerator;
-        $this->allowedGitHubOrganization = $allowedGitHubOrganization;
+    public function __construct(
+        private ClientRegistry $clientRegistry,
+        private EntityManagerInterface $em,
+        private UrlGeneratorInterface $urlGenerator,
+        private string $allowedGitHubOrganization
+    ) {
         $this->httpClient = HttpClient::create();
```

Comment:
Let's inject the client here too: If symfony inject the client, it will be "monitored" in the profiler and in the logger